Tests/Configurations/TestRemoteConfiguration.swift

```diff
@@ -306,53 +306,67 @@ class TestRemoteConfiguration: XCTestCase {
         wait(for: [expectation], timeout: 10)
     }
     
-#endif
-    
-    // TODO: Replace LSNocilla as it's unreliable and unsupported. It causes this test failure.
-    /*
-    - (void)testConfigurationProvider_justRefresh_downloadHigherConfigVersionThanCached_doUpdate {
+    func testUsesDefaultConfigurationIfTheSameConfigurationVersionAsFetched() {
         // prepare test
-        NSString *endpoint = @"https://fake-snowplow.io/config.json";
-        SPRemoteConfiguration *remoteConfig = [[SPRemoteConfiguration alloc] initWithEndpoint:endpoint method:SPHttpMethodGet];
-
-        SPConfigurationCache *cache = [[SPConfigurationCache alloc] initWithRemoteConfiguration:remoteConfig];
-        [cache clearCache];
-        SPConfigurationBundle *bundle = [[SPConfigurationBundle alloc] init];
-        bundle.networkConfiguration = [[SPNetworkConfiguration alloc] initWithEndpoint:@"endpoint"];
-        SPFetchedConfigurationBundle *cached = [[SPFetchedConfigurationBundle alloc] init];
-        cached.schema = @"http://iglucentral.com/schemas/com.snowplowanalytics.mobile/remote_config/jsonschema/1-0-0";
-        cached.configurationVersion = 1;
-        cached.configurationBundle = @[bundle];
-        [cache writeCache:cached];
-        [NSThread sleepForTimeInterval:5]; // wait to write on cache
-
-        SPConfigurationProvider *provider = [[SPConfigurationProvider alloc] initWithRemoteConfiguration:remoteConfig];
-        XCTestExpectation *expectation = [XCTestExpectation new];
-        [provider retrieveConfigurationOnlyRemote:NO onFetchCallback:^(SPFetchedConfigurationBundle * _Nonnull fetchedConfigurationBundle) {
-            [expectation fulfill];
-        }];
-        [self waitForExpectations:@[expectation] timeout:5];
-
-        [[LSNocilla sharedInstance] start];
-        stubRequest(@"GET", endpoint)
-        .andReturn(200)
-        .withHeaders(@{@"Content-Type": @"application/json"})
-        .withBody(@"{\"$schema\":\"http://iglucentral.com/schemas/com.snowplowanalytics.mobile/remote_config/jsonschema/1-1-0\",\"configurationVersion\":2,\"configurationBundle\":[]}");
+        let endpoint = "https://fake-snowplow.io/config.json"
```

Comment:
Good point, it's better to generate different inputs rather than using the same in all tests. I had ChatGPT generate a few fake URI and added a function that automatically picks one from them.